movement/watch_faces/clock/clock_face.c

```diff
@@ -57,7 +53,6 @@ typedef struct {
 } clock_state_t;
 
 static bool clock_is_in_24h_mode(movement_settings_t *settings) {
-    if (CLOCK_FACE_24H_ONLY) { return true; }
```

Comment:
In 24h only mode, this line reduces to:

    if (1) { return true; }

The purpose of this line is to allow the compiler to delete the function and replace it with `true` everywhere the code must branch into either 12h or 24h mode. The intention was to facillitate dead code elimination of the 12h mode code.

Can you confirm if this property still holds without the preprocessor definition?